wordpress-mobile/WordPress-iOS
# Jetpack migration: Wire up export function to the UI in WordPress

Refs #19668 

- Adds a `ContentMigrationCoordinator` to be called from the WP side. This class handles checking the migration condition and calling the relevant migration methods.
- Wire up the coordinator object to the Jetpack overlay sheets.
- Wire up the coordinator to App Delegate, to be called once upon app launch. If the export process is successful, it will not be called again unless the app is reinstalled.

## To test:

### Unit test
- Ensure that the unit tests are passing.

### Manual test

_Note: To test this manually, you will need some tools to help verify the exported data. The easiest would be to run the app in a simulator and use a simulator observer tool such as [SimSim](https://github.com/dsmelov/simsim) so you can easily access the shared folder used by the app. And also [SQLite Browser](https://sqlitebrowser.org/) to inspect the copied database file._

#### Scenario 1: Export once

- Launch the WordPress app.
- Login to an account that has blogs.
- Enable the content migration and the Jetpack-powered bottom sheet feature flags.
- Relaunch the app.
- The export function should get called here once.
- Verify that the data export process succeeds:
  - Open the `group.org.wordpress` App Group folder.
  - Verify that the database file has been backed up.
  - Verify that the user defaults are backed up.

#### Scenario 2: Don't export for the second time during app launch

- Using the previous session, add a self-hosted app.
- Relaunch the app.
- The export function should NOT be called again here (provided that the first export succeeded). To verify:
  - Open the `group.org.wordpress` App Group folder.
  - Open the database backup file with SQLite Browser.
  - Verify that the self-hosted site is not there by looking at the records in the `Blog` schema.

#### Scenario 3: Tapping on the Jetpack button exports WP data.

- While logged-in, add a self-hosted site.
- Go to any section with the Jetpack powered badge, and tap it.
- Tap on the Try the new Jetpack app.
- It doesn't matter whether JP is installed or not, the data export will be performed.
- Verify that the data has been exported:
  - Go to the `group.org.wordpress` App Group folder.
  - Open the database file.
  - Verify that the self-hosted site is in the `Blog` schema.

## Regression Notes
1. Potential unintended areas of impact
Should be none. It's not released yet.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Added unit tests.

3. What automated tests I added (or what prevented me from doing so)
Added tests for `ContentMigrationCoordinator`.

PR submission checklist:
